spec/manifolds/cli_spec.rb

```diff
@@ -3,86 +3,102 @@
 require_relative "../../lib/manifolds/cli"
 require "fileutils"
 require "logger"
+require "fakefs/spec_helpers"
 
 RSpec.describe Manifolds::CLI do
+  include FakeFS::SpecHelpers
+
   let(:project_name) { "commerce" }
   let(:sub_project_name) { "Pages" }
   let(:null_logger) { Logger.new(File::NULL) }
-  let(:bq_service) { instance_double("Manifolds::Services::BigQueryService", "commerce") }
+
+  before do
+    FakeFS do
+      # Set up any template files that need to exist
+      FileUtils.mkdir_p("#{File.dirname(__FILE__)}/../../lib/manifolds/templates")
+      File.write("#{File.dirname(__FILE__)}/../../lib/manifolds/templates/config_template.yml", "vectors:\nmetrics:")
+      File.write("#{File.dirname(__FILE__)}/../../lib/manifolds/templates/vector_template.yml", "attributes:")
+    end
+  end
 
   describe "#init" do
     subject(:cli) { described_class.new(logger: null_logger) }
 
-    before do
-      allow(FileUtils).to receive(:mkdir_p)
-      allow(File).to receive(:open)
-      cli.init(project_name)
-    end
+    context "when initializing a new project" do
+      before { cli.init(project_name) }
 
-    it "creates the projects directory" do
-      expect(FileUtils).to have_received(:mkdir_p).with("./#{project_name}/projects")
+      it "creates a project directory structure" do
+        expect(Dir.exist?("./#{project_name}")).to be true
+        expect(Dir.exist?("./#{project_name}/projects")).to be true
+        expect(Dir.exist?("./#{project_name}/vectors")).to be true
```

Comment:
Either split this into two tests (my preference), or wrap them in an `aggregate_failures` block.

Also, `expect(Pathname("some/path")).to be_directory` reads a lot better.

```ruby
it "creates a 'projects' directory for the project" do
  cli.init(project_name)
  expect(Pathname(".#{project_name}/projects")).to be_directory
end

it "creates a 'vectors' directory for the project" do
  cli.init(project_name)
  expect(Pathname(".#{project_name}/vectors")).to be_directory
end
```

```ruby
it "creates 'projects' and 'vectors' directories for the project" do
  cli.init(project_name)
  aggregate_failures do
    expect(Pathname(".#{project_name}/projects")).to be_directory
    expect(Pathname(".#{project_name}/vectors")).to be_directory
  end
end
```